### ISSUE TITLE
next_holidays method returns empty arrays on May at jp region.

### ISSUE THREAD

**Issue by ttwo32** [holidays/holidays#194]
At jp regions, next_holidays method returns empry arrays on May.

```
irb(main):002:0> Date.today   
=> #<Date: 2016-05-22 ((2457531j,0s,0n),+0s,2299161j)>
irb(main):002:0> Holidays.next_holidays(1,[:jp])
=> []
```

I have surveyed a cause.
Problem line : https://github.com/holidays/holidays/blob/master/lib/holidays.rb#L154

When next_holidays was called, i have made from_date's date_driver only.
In my mind, we should make end_date's date_driver like between method,
and end_date's sholud be optional param, defaulted to `from_date >> 12` .

What are your thoughts?

**Comment by ptrimble** [comment]
As we discussed via DMs I see the problem but I want to play with it a bit...I'm not sure I want to just add the `from_date`. 

I'll try to get a PR up this weekend. Thanks!

**Comment by ptrimble** [comment]
@ttwo32 Okay, I _totally_ see what you mean. Sorry, language barrier there. Your solution was the smart one, I went ahead and implemented it. 

I just went ahead and did 12 months out without requiring anyone to submit anything new. I don't think this will have a performance impact. If it turns out we do then we can make the algorithm smarter.

Let me know what you think! 

https://github.com/holidays/holidays/pull/196

**Comment by ttwo32** [comment]
Thank you for refactoring.
I don't think this will have a performance impact.
Thanks!! 😄

**Comment by ptrimble** [comment]
Great, once I release this change (it'll be version 4.2.0) I'll close out this issue. Thanks!

**Comment by ptrimble** [comment]
This has been released as v4.2.0. Thanks again @ttwo32!
